packages/js/product-editor/src/blocks/regular-price/edit.tsx

```diff
@@ -64,6 +73,15 @@ export function Edit( {
 		'wp-block-woocommerce-product-regular-price-field'
 	) as string;
 
+	const [ productAttributes ] = useEntityProp< Product[ 'attributes' ] >(
+		'postType',
+		'product',
+		'attributes'
+	);
+	const isDisabled =
+		hasAttributesUsedForVariations( productAttributes ) &&
+		isSelectedTabApplicableForOptionsNotice( context?.selectedTab );
```

Comment:
It doesn't make to much sense to me that the pricing fields are disabled on the Pricing tab, but not in the General tab. Given I can still add a price in the General tab, this just causes confusing behaviour (both should be disabled in this case, or neither ).
Also if I add pricing through the general tab I can still schedule a sale, as this isn't disabled.

Reply:
We plan to remove the pricing fields from the General tab anyway, and I'm not sure if we could do it as part of this issue?

Reply:
As for disabling fields... to be honest, we decided against this while working on the first version of the form. Are you aware of any changes in this area? I think it's okay to just show the message at the top. There are several situations when the user may want to use the parent product's fields and disabling them could hurt their workflows.

Reply:
> We plan to remove the pricing fields from the General tab anyway, and I'm not sure if we could do it as part of this issue?

We probably want to do that as part of a separate issue/PR.

> Are you aware of any changes in this area?

I am not aware of any changes in this area, although I wasn't even aware of the disabling fields part 🤷‍♂️ I am good with not going with the disabling part for the reasons you mention.
@octaedro could you remove that part from this PR?

Reply:
No, it's my fault! We should have made that clearer after we changed the AC. We'll do better next time. 🙏